pybossa/cache/projects.py

```diff
@@ -630,6 +630,21 @@ def get_from_pro_user():
         projects.append(project)
     return projects
 
+def get_recently_updated_projects():
+    """Return the list of published projects that has task creations in last 3 months."""
+
+    sql = text('''SELECT t.project_id, p.short_name FROM task t, project p
+               GROUP BY t.project_id, p.short_name, p.id
+               HAVING t.project_id = p.id
+               AND p.published = true
+               AND TO_DATE(max(t.created), 'YYYY-MM-DD"T"HH24:MI:SS.US') >= NOW() - '3 months' :: INTERVAL;''')
+    results = db.slave_session.execute(sql)
+    projects = []
+    for row in results:
+        project = dict(id=row.project_id, short_name=row.short_name)
+        projects.append(project)
+    return projects
```

Comment:
Any idea how long it will long? My guess is that, this will be long running SQL as well because anytime we join task or task_run table, it won't be fast. Nevertheless, we can properly cache it.

Reply:
There's no task run table involved in this query. We're looking for projects with task creations in last 3 months.

Reply:
My bad(Guess I'm too worried about those two tables lol). Still a good idea to know the running time and cache is necessary to reduce the DB load.

Reply:
It took 10 seconds to obtain the data. Since its select query, I'm hopeful that this will run in parallel along with other  sql queries and need not to be dependent on outcome of other queries.
```
SELECT t.project_id, p.short_name FROM task t INNER JOIN project p ON t.project_id = p.id GROUP BY t.project_id, p.short_name, p.id HAVING p.published = true AND TO_DATE(max(t.created), 'YYYY-MM-DD"T"HH24:MI:SS.US') >= NOW() - '3 months'  :: INTERVAL;
Time: 10831.329 ms (00:10.831)
```